src/attr_file.c

```diff
@@ -397,7 +397,7 @@ bool git_attr_fnmatch__match(
 
 		/* for attribute checks or root ignore checks, fail match */
 		if (!(match->flags & GIT_ATTR_FNMATCH_IGNORE) ||
-			path->basename == path->path)
+			path->basename == relpath)
```

Comment:
This is not really a fault of yours, as you simply mirror the previous style, but: do we really want to compare the pointers here? I bet this should be a `strcmp` instead. And if not, we should at least add a comment here to explain what's going on.